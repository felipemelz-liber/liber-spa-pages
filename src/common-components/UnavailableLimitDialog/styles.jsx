import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
