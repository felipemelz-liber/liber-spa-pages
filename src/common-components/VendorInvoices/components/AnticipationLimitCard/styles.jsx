import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
