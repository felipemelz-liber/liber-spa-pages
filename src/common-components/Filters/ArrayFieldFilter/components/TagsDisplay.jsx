import React from 'react';
import PropTypes from 'prop-types';
import { GridContainer, Grid, Tag, PlusText } from './TagsDisplay.styles';

const hasTags = tagsArray => tagsArray.length > 0;

const TagsDisplay = ({ tags, tagsPerLine, tagsThreshold }) => {
  let extraTags = [];
  let displayTags = tags;
  const columnSize = 12 / tagsPerLine;

  if (tags.length > tagsThreshold) {
    displayTags = tags.slice(0, tagsThreshold - 1);
    extraTags = tags.slice(tagsThreshold - 1);
  }
  return (
    <GridContainer spacing={2}>
      {hasTags(displayTags)
        ? displayTags.map(invoiceNumber => (
            <Grid key={invoiceNumber} columnSize={columnSize}>
              <Tag>{invoiceNumber}</Tag>
            </Grid>
          ))
        : null}
      {hasTags(extraTags) ? (
        <Grid columnSize={columnSize}>
          <PlusText>{`+ ${extraTags.length}`}</PlusText>
        </Grid>
      ) : null}
    </GridContainer>
  );
};

TagsDisplay.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string.isRequired),
  tagsThreshold: PropTypes.number.isRequired,
  tagsPerLine: PropTypes.number.isRequired,
};

TagsDisplay.defaultProps = {
  tags: [],
};

export default TagsDisplay;
