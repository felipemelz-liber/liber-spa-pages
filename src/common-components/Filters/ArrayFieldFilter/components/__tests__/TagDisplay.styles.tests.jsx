import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import * as styles from '../TagsDisplay.styles';

describe('TagsDisplay styled components test', () => {
  configure({ adapter: new Adapter() });

  const { Grid, GridContainer, ...Components } = styles;

  Object.keys(Components).forEach(componentName => {
    const Component = Components[componentName];
    it(`should render ${componentName} correctly`, () => {
      const wrapper = shallow(<Component />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  it('should render GridContainer correctly', () => {
    const wrapper = shallow(<GridContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render GridContainer with spacing correctly', () => {
    const wrapper = shallow(<GridContainer spacing={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Grid correctly', () => {
    const wrapper = shallow(<Grid />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Grid correctly with columnSize', () => {
    const wrapper = shallow(<Grid columnSize={2} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render GridContainer with Grid correctly', () => {
    const wrapper = shallow(
      <GridContainer spacing={4}>
        <Grid />
      </GridContainer>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
