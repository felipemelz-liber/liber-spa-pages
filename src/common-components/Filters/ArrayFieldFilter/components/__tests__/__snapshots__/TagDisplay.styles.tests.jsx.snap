// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsDisplay styled components test should render Grid correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  flex-basis: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TagsDisplay styled components test should render Grid correctly with columnSize 1`] = `
<StyledComponent
  columnSize={2}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  flex-basis: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TagsDisplay styled components test should render GridContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-wrap: wrap;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TagsDisplay styled components test should render GridContainer with Grid correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-wrap: wrap;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  spacing={4}
>
  <styled.div />
</StyledComponent>
`;

exports[`TagsDisplay styled components test should render GridContainer with spacing correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-wrap: wrap;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  spacing={4}
/>
`;

exports[`TagsDisplay styled components test should render PlusText correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.43;
  letter-spacing: 0.02px;
  text-align: left;
  color: #798e9b;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TagsDisplay styled components test should render Tag correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  margin-right: 0px;
  color: #798e9b;
",
        ],
      },
      "displayName": "Styled(Tag)",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "children": [Function],
        "closable": [Function],
        "height": [Function],
        "onClose": [Function],
        "variant": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "children": "",
          "closable": false,
          "height": 28,
          "onClose": [Function],
          "variant": "default",
        },
        "displayName": "Tag",
        "propTypes": Object {
          "children": [Function],
          "closable": [Function],
          "height": [Function],
          "onClose": [Function],
          "variant": [Function],
        },
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
